deliver/lib/deliver/upload_metadata.rb

```diff
@@ -400,6 +401,18 @@ def assign_defaults(options)
       end
     end
 
+    def detect_existing_app_store_languages(options)
+      app = options[:app]
+
+      enabled_languages = detect_languages(options)
```

Comment:
Calling `detect_languages` again after L366 doesn't look like fair here. Shouldn't this be a part of `detect_languages` and return the same values to all the call sites of `detect_languages`?

We use `detect_languages` and the variable `enabled_languages` here in `UploadMetadata#upload` as well. I'm afraid that a small discrepancy in `enabled_languages` between here and there might cause a little confusion later on. I think merging `detect_existing_app_store_languages` into `detect_languages` would slightly simplify entire logic by freeing from you to think the differences.

https://github.com/fastlane/fastlane/blob/678966c4175a9e51434dae97a5b147bcb7dfc8d9/deliver/lib/deliver/upload_metadata.rb#L89-L92 

I might be wrong and it might not that be easy though😅 What do you think?

Reply:
Ohhhhhh... This _IS_ super weird 😱  Yeah yeah yeah, I think you are right 😇 I will try to rework!